src/gamestate/system_state.cpp

```diff
@@ -2334,6 +2335,9 @@ void state::load_scenario_data() {
 						auto content = view_contents(*opened_file);
 						parsers::token_generator gen(content.data, content.data + content.file_size);
 						parsers::parse_country_history_file(gen, err, new_context);
+						
+						for(auto& e : new_context.pending_decisions)
```

Comment:
this loop could be done faster with just `pending_decisions = std::move(new_context.pending_decisions));` which will transfer the internal buffer of one vector to the other, no copies or new allocations required

Reply:
Oh, I see, you have to do this for each file. In that case, you could put a reference to the outer vector in the context for some marginal savings